task_sdk/src/airflow/sdk/execution_time/supervisor.py

```diff
@@ -265,7 +270,13 @@ class WatchedSubprocess:
     _terminal_state: str | None = None
     _final_state: str | None = None
 
-    _last_heartbeat: float = 0
+    _last_successful_heartbeat: float = 0
+    _last_heartbeat_attempt: float = 0
```

Comment:
```suggestion
    _last_successful_heartbeat: float = attrs.Field(default=0, init=False)
    _last_heartbeat_attempt: float = attrs.Field(default=0, init=False)
```

to make it not show up as args in init